utils/kafka/src/main/java/io/apicurio/registry/utils/kafka/KafkaUtil.java

```diff
@@ -60,41 +63,61 @@
         try (Admin admin = Admin.create(properties)) {
-            createTopics(admin, topicNames);
+            createTopics(admin, topicNames, topicConfig);
         }
     }
+    public static void createTopics(Properties properties, Set<String> topicNames) {
+        createTopics(properties, topicNames, null);
+    }
 
     /**
      * Create topics with sensible defaults.
      *
      * @param admin the Kafka admin to use
      * @param topicNames topics to create, if they don't exist
+     * @param topicConfig the config to use for the new topic
      */
-    public static void createTopics(Admin admin, Set<String> topicNames) {
+    public static void createTopics(Admin admin, Set<String> topicNames, Map<String, String> topicConfig) {
+        if (topicConfig == null) {
+            topicConfig = new HashMap<>();
+        }
+
         Set<String> topics = result(admin.listTopics().names());
         List<NewTopic> topicsToCreate = new ArrayList<>();
         for (String topicName : topicNames) {
-            createTopic(admin, topics, topicsToCreate, topicName);
+            NewTopic newTopic = createTopic(admin, topics, topicName, topicConfig);
+            if (newTopic != null) {
+                topicsToCreate.add(newTopic);
+            }
         }
         if (topicsToCreate.size() > 0) {
             result(admin.createTopics(topicsToCreate).all());
         }
     }
-
-    private static void createTopic(Admin admin, Set<String> topics, List<NewTopic> topicsToCreate, String topicName) {
-        if (!topics.contains(topicName)) {
+    public static void createTopics(Admin admin, Set<String> topicNames) {
+        createTopics(admin, topicNames, null);
+    }
+    
+    private static NewTopic createTopic(Admin admin, Set<String> existingTopicNames, String topicName, final Map<String, String> topicProperties) {
+        if (!existingTopicNames.contains(topicName)) {
             KafkaFuture<NewTopic> newTopicKF = admin.describeCluster().nodes().thenApply(nodes -> {
-                log.info("Creating new topic: {}", topicName);
-                int rf = Math.min(3, nodes.size());
-                int minISR = Math.max(rf - 1, 1);
-                return new NewTopic(topicName, 1, (short) rf)
-                        .configs(Collections.singletonMap(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(minISR)));
+                log.info("Creating new Kafka topic: {}", topicName);
+                int replicationFactor = Math.min(3, nodes.size());
+                if (topicProperties.containsKey("replication.factor")) {
+                    replicationFactor = Integer.parseInt(topicProperties.get("replication.factor"));
+                }
+                int minimumInSyncReplicas = Math.max(replicationFactor - 1, 1);
+                topicProperties.putIfAbsent(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(minimumInSyncReplicas));
```

Comment:
Where is topicProperties used?

Reply:
Whoops!  It's supposed to be passed to `.configs()`.  Thanks for catching this.  Bonehead mistake.